mvcc/kvstore.go

```diff
@@ -271,9 +271,13 @@ func (s *store) Compact(rev int64) (<-chan struct{}, error) {
 			return
 		}
 		close(ch)
-		s.keepMu.Lock()
-		s.keep = nil
-		s.keepMu.Unlock()
+		s.revMu.RLock()
+		if rev == s.compactMainRev {
+			s.keepMu.Lock()
+			s.keep = nil
```

Comment:
If I understand how keep works, the keep from this compaction only contains revisions between compact rev of last compaction to this compaction rev. If last compaction hasn't finished then, then the current keep does not include the keep from last compaction. Hence, HashKV will skip revisions from the keep of last compaction which is not what we want.